### ISSUE TITLE
idone -[yd] should list tasks, not wait for input

### ISSUE THREAD

**Issue by wizonesolutions** [pjf/idonethis-perl#10]
So often when I type `idone -y` or `idone -d`, I actually just want to list out tasks from a previous day. Naturally, this puts me into input mode automatically, since that is the default behavior.

Is it possible to set default action per item? Mind that I **do** want to go into entry mode with just `idone`, but when specifying a date switch it'd be nice to list by default unless I explicitly specified `-c`.

It would be extra-cool if it could detect if I entered a string at the end of my incantation containing `-d` or `-y` and sent a single entry to iDoneThis instead of listing. However, that deviance in behavior might not be intuitive, so it's just a hope.
##

**Comment by pjf** [comment]
I'm not sure I understand your last paragraph. Don't we already do that?

`idone -y "Did something yesterday"`

**Comment by pjf** [comment]
Or do you mean you'd like to be able to do:

`idone -ly "Did something yesterday?"`

or even:

`idone -lcy "Did something yesterday?"`

They might work already, I haven't tested them. :)

**Comment by wizonesolutions** [comment]
Nah, just the normal incantations for adding a single done to a particular date or `-y`. What I meant was that if I supply a terminal string argument that the alternate behavior of simply listing dones for a particular date should not occur. Instead, it should add just add the done like it currently does. Was that clearer?

**Comment by pjf** [comment]
Nope! I'm still confused!  When I type:

`idone -y "Cleaned ALL the things`

It just adds that to yesterday's todo list. It doesn't list anything, or wait for me to add more items. That sounds like the behaviour you're expecting, yes?

(Although if you're getting something different, try upgrading with `cpanm WebService::Idonethis`, it might be that there was a bug and I've since fixed it...)

**Comment by wizonesolutions** [comment]
Yeah, that explains the behavior I expect when supplying a string. So the
first part of my request was to be able just to say `idone -y` and get a
list of yesterday's tasks. If I want to stdin-input then I can `idone -yc`
:)

**Comment by pjf** [comment]
Ohhhhhh! Yes, `idone -y` and `idone -d` should definitely just list tasks.  This is definitely a bug.

**Comment by pjf** [comment]
Oh wait, maybe they shouldn't, because `idone` without arguments at all doesn't list tasks, it accepts them as input.

(I've actually become such a heavy user of `idone -lc` and `idone -lcy` that I've stopped using the defaults.)

If a plain `idone` simply listed tasks completed today (ie, the same as `idone -c`) then that would make everything consistent.  I don't think I have any moral objections to that...

I'm also tempted to make a `defaults` section in the .rc file which can override the default behaviour (when absolutely no arguments are given).  I almost _always_ want `idone -lc`. However I'm wary this might be over-featuring things.

**Comment by wizonesolutions** [comment]
Yeah, striking the balance. I like both idone going into stdin mode for
today and any date flags given without -c listing. I am ok with idone
simply listing if it's easier though. -c is easy to tack on when I want to
dump them in. I tend to do idone "whatever" several times throughout the
day. I use -c to catch up.

**Comment by pjf** [comment]
So, part-way related to this, I'm about to push a change so that terminal arguments are always sent as dones. This means you can do `idone -l Frobnicated the Hoodibbits` and it will add `Frobnicated the Hoodibbits` to your done list, and then show you everything you've done today.
